### PR TITLE
feat: clear conversation content

### PR DESCRIPTION
**Pull request by trOnk12**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

-Add support for clearing the content

image

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1144/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1144/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1144/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1144/6/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/HomeViewModel.kt [hunk 1]
```diff
@@ -114,14 +115,17 @@ data class HomeState(
...
 sealed class HomeSnackbarState {
+    object None : HomeSnackbarState()
+    data class ClearConversationContentSuccess(val isGroup: Boolean) : HomeSnackbarState()
+    data class ClearConversationContentFailure(val isGroup: Boolean) : HomeSnackbarState()
+
     class SuccessConnectionIgnoreRequest(val userName: String) : HomeSnackbarState()
     object MutingOperationError : HomeSnackbarState()
     object BlockingUserOperationError : HomeSnackbarState()
-    class BlockingUserOperationSuccess(val userName: String) : HomeSnackbarState()
+    data class BlockingUserOperationSuccess(val userName: String) : HomeSnackbarState()
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/HomeViewModel.kt:125]: Any reason declaring these 2 classes as `data classes` but not for `SuccessConnectionIgnoreRequest` ?

**Review by gongracr — APPROVED** [review]
LGTM 🧑‍🍳 🥟

File: app/src/main/kotlin/com/wire/android/ui/home/HomeViewModel.kt (hunk 1 above)
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/HomeViewModel.kt:125]: no idea I was not responsible for the other ones

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1144/7/) **failed**.

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1144/8/) **failed**.

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-1144/9/) **failed**.

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-1144/10/) **failed**.

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-1144/11/) **failed**.

**Comment by AndroidBob** [comment]
Build [12](https://10.10.124.134/job/wire-android-reloaded/job/PR-1144/12/) **failed**.

**Comment by AndroidBob** [comment]
Build [13](https://10.10.124.134/job/wire-android-reloaded/job/PR-1144/13/) **failed**.

**Comment by AndroidBob** [comment]
Build [14](https://10.10.124.134/job/wire-android-reloaded/job/PR-1144/14/) **failed**.

**Comment by AndroidBob** [comment]
Build [15](https://10.10.124.134/job/wire-android-reloaded/job/PR-1144/15/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4988905)(4988905)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4988905)(4988905)-dev-release.apk)

**Pull request merged**